src/main/java/seedu/address/logic/commands/AssignCommand.java

```diff
@@ -0,0 +1,68 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.List;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.CommandHistory;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.person.Guest;
+import seedu.address.model.room.RoomNumber;
+
+/**
+ * Assigns a guest to a room.
+ */
+public class AssignCommand extends Command {
+
+    public static final String COMMAND_WORD = "assign";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Assigns a guest to a room.\n"
+            + "Parameters: GUEST_NAME ROOM_NUMBER\n"
+            + "Example: " + COMMAND_WORD + " Thomas 056";
+
+    public static final String MESSAGE_ASSIGN_GUEST_SUCCESS =
+            "Assigned Guest: %1$s to Room %1$s";
+
+    private final Index targetIndex;
```

Comment:
KIV. Not sure why this is needed. `Assign` command does not currently make use of any `Index`.